oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/JwtOAuth2TokenCustomizer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.authorization.token;
+
+import org.springframework.security.oauth2.jose.JoseHeader;
+import org.springframework.security.oauth2.jwt.JwtClaimsSet;
+
+/**
+ * An extension point to customize generated Jwt token.
+ *
+ * @see JwtAccessTokenOAuth2TokenIssuer
+ * @see OAuth2TokenIssuer
+ * @see org.springframework.security.oauth2.jwt.Jwt
+ *
+ * @author Alexey Nesterov
+ * @since 0.0.4
+ */
+@FunctionalInterface
+public interface JwtOAuth2TokenCustomizer {
```

Comment:
I would prefer to introduce a general interface that would suit customizing the claims of a `Jwt` and opaque access token, e.g. `OAuth2TokenCustomizer`. We might need to introduce a hierarchy for customizing the `Jwt` header, something like `interface JwtTokenCustomizer extends OAuth2TokenCustomizer` and provides the additional operation for customizing the headers.